### PR TITLE
Refactor/UI ux enhancement p4

### PR DESCRIPTION
**Pull request by phucvarmeta**

### CODE REVIEW

File: packages/web/src/models/pool/mapper/pool-mapper.ts [hunk 1]
```diff
@@ -95,15 +95,15 @@ export class PoolMapper {
       tokenA,
       tokenB,
       feeTier: feeTierInfo?.type || "NONE",
-      apr: apr !== null ? apr.toString() : "",
+      apr: (apr ?? "").toString(),
```
Comment (jinoosss) [inline packages/web/src/models/pool/mapper/pool-mapper.ts:98]: The modification of the APR should be done in the `fromResponse` `detailFromResponse` functions, not in this part.
You can use `totalApr` instead of `apr`. 

```
public static fromResponse(pool: PoolResponse): PoolModel {
    const id = pool.id ?? makeId(pool.poolPath);
    return {
      ...pool,
      id,
      incentiveType: pool.incentiveType as INCENTIVE_TYPE,
      rewardTokens: pool.rewardTokens || [],
      apr: pool.totalApr,
      bins: pool?.bins || [],
      bins40: pool?.bins40 || [],
      liquidity: pool.liquidity,
      allTimeVolumeUsd: pool.allTimeVolumeUsd,
      price: Number(pool.price),
    };
  }
```

**Review by jinoosss — COMMENTED** [review]
I left a few small comments.

File: packages/web/src/containers/swap-container/SwapContainer.tsx [hunk 2]
```diff
@@ -68,17 +68,20 @@ const SwapContainer: React.FC = () => {
       tokens.find(token => token.path === query.from) || null;
     const currentTokenB =
       tokens.find(token => token.path === query.to) || null;
-    const direction: SwapDirectionType =
-      query.direction === "EXACT_OUT" ? "EXACT_OUT" : "EXACT_IN";
+    const direction = query.direction as SwapDirectionType;
+    // const tokenAAmountQuery = (query.token_a_amount ?? "") as string;
+    // const tokenBAmountQuery = (query.token_b_amount ?? "") as string;
```
[the remark this answers is not in the document]
Comment (phucvarmeta) [reply packages/web/src/containers/swap-container/SwapContainer.tsx:73]: I will use these line later

File: packages/web/src/components/common/line-graph/LineGraph.tsx [hunk 3]
```diff
@@ -354,6 +354,20 @@ const LineGraph: React.FC<LineGraphProps> = ({
         .toNumber();
     };
 
+    if (hasOnlyOnePoint) {
+      const onlySingleData = mappedDatas[0];
+
+      // const x = BigNumber(time - minTime)
+      // .multipliedBy((width) - baseLineNumberWidthComputation)
+      // .dividedBy(maxTime - minTime)
+      // .toNumber();
```
[the remark this answers is not in the document]
Comment (phucvarmeta) [reply packages/web/src/components/common/line-graph/LineGraph.tsx:363]: i will use these line afterward, please ignore this for now

File: packages/web/src/models/pool/mapper/pool-mapper.ts (hunk 1 above)
Comment (phucvarmeta) [reply packages/web/src/models/pool/mapper/pool-mapper.ts:98]: I will working this together with api migration

File: packages/web/src/models/pool/mapper/pool-mapper.ts [hunk 4]
```diff
@@ -95,15 +95,15 @@ export class PoolMapper {
       tokenA,
       tokenB,
       feeTier: feeTierInfo?.type || "NONE",
-      apr: apr !== null ? apr.toString() : "",
```
Comment (jinoosss) [inline packages/web/src/models/pool/mapper/pool-mapper.ts:98]: It's probably a good idea to revert the changes on this line.

File: packages/web/src/models/pool/mapper/pool-mapper.ts [hunk 5]
```diff
@@ -38,15 +38,15 @@ export class PoolMapper {
...
```
Comment (jinoosss) [inline packages/web/src/models/pool/mapper/pool-mapper.ts:41]: It's probably a good idea to revert the changes on this line.

**Review by jinoosss — CHANGES_REQUESTED** [review]
Please check the comments to resolve any build errors.

File: packages/web/src/models/pool/mapper/pool-mapper.ts (hunk 5 above)
Comment (phucvarmeta) [reply packages/web/src/models/pool/mapper/pool-mapper.ts:41]: it's some issue in fake data, i cheked it

File: packages/web/src/models/pool/mapper/pool-mapper.ts (hunk 4 above)
Comment (phucvarmeta) [reply packages/web/src/models/pool/mapper/pool-mapper.ts:98]: it's some issue in fake data, i cheked it

**Review by jinoosss — APPROVED** [review]
